### PR TITLE
Issue440 properties page work

### LINKED ISSUE #440: Properties Page Bug + Link to Edit Property page
**Purpose**
**Fix a bug:** the Manage Properties page has a search capability that works to reduce the active items in the table, but there is no way to go back to the complete list. To replicate the bug:
1. Search for a property and press enter
2. Note that the list of properties shrinks
3. Click the "X" in the search bar or delete all search text and press enter again
4. The list of properties does not return to its original length

**Add a link to edit a property.** 
- Pages 49 and 50 of the [FIGMA designs](https://drive.google.com/file/d/1YqboQogczYm1HkyRqEtVSzeQ61T9hWU2/view) show a page for editing properties, and there needs to be a way to get to that view from the Manage Properties page

**Add a way to archive one or more properties.** 
- The checkboxes on the Manage Properties page are intended for selecting a group of properties and performing a bulk action on those properties. One bulk action that has been identified (so far): "Archive". 

**Add a way to view archived properties.** 
- Similar to page 36 where the [FIGMA designs](https://drive.google.com/file/d/1YqboQogczYm1HkyRqEtVSzeQ61T9hWU2/view) show a toggle for viewing archived property managers, the view properties page needs to have a toggle for viewing archived properties (or not)

**Proposed Solution**
- [x] When the search bar is used, it should filter the rows based on all visible properties that contain the search criteria.
- [x] The search bar needs an onChange event while typing so the results are filtered in real-time. When the user erases their search criteria (or clicks the "X" on the right hand side of the search bar), it should reset to the original unfiltered data set.
- [x] Make the name of the property clickable as a link to an edit property page. The edit property page does not yet exist, so the URL will change but the page will remain unchanged
- [x] Add an action button called "Archive Property". By default, this button is disabled and it only becomes active when one or more properties have their checkboxes selected.
- [x] Clicking the Archive button will submit a request to the backend. This endpoint needs to be defined. A proposal is to use a POST request to `/api/properties/archive/<id>` and send the propertyIDs.
  - It may be possible to send an array of ids through the url for this endpoint. Test this with the backend
  - Suggest any necessary changes to the backend (tweak for array compatibility or overhaul) as a new issue
- [x] After the archive request is complete, refresh the list of properties (GET `/api/properties`) and update the list on the page
- [x] Add a switch/toggle (a checkbox styled as a switch) for viewing archived properties. Build this as a re-usable component because other pages will need a similar feature.

**Please make sure you've attempted to meet the following coding standards**
- [x] Code builds successfully
- [x] Code has been tested and does not produce errors
- [x] Code is readable and formatted
- [x] There isn't any unnecessary commented-out code
- [x] There aren't any unnecessary commits or files changed (shouldn't touch the /stashed folder unless the issue requests it)

**Mockup Screenshot or Page Number**
Page 14 of the [FIGMA designs](https://drive.google.com/file/d/1YqboQogczYm1HkyRqEtVSzeQ61T9hWU2/view) shows the Manage Properties page

Page 34 of the [FIGMA designs](https://drive.google.com/file/d/1YqboQogczYm1HkyRqEtVSzeQ61T9hWU2/view) shows that multiple properties can be selected via the checkboxes

Pages 49 and 50 of the [FIGMA designs](https://drive.google.com/file/d/1YqboQogczYm1HkyRqEtVSzeQ61T9hWU2/view) show a view for editing properties

Page 36 of the [FIGMA designs](https://drive.google.com/file/d/1YqboQogczYm1HkyRqEtVSzeQ61T9hWU2/view) shows an example of the toggle that can be used to show/hide archived items

### PR DESCRIPTION
**Pull request by DaveTrost**
### What issue is this solving?
Addresses issue #440

### Any helpful knowledge/context for the reviewer?
Is a `npm install` necessary? No
Any special requirements to test? No

### Please make sure you've attempted to meet the following coding standards
- [x] Code builds successfully
- [x] Code has been tested and does not produce errors
- [x] Code is readable and formatted
- [x] There isn't any unnecessary commented-out code
- [x] There aren't any unnecessary commits or files changed (shouldn't touch the /stashed folder unless the issue requests it)

If you're having trouble meeting this criteria, feel free to reach out on Slack for help!

### CODE REVIEW

**Comment by DaveTrost** [comment]
I opened a backend PR related to the functionality in this branch: codeforpdx/dwellinglybackend/pull/166
Ideally, the backend PR can be closed before this one so I can add to this branch with the code that makes a request to the new API route

File: src/components/ShowHideSwitch/index.js
Comment (Chris-Boe) [inline src/components/ShowHideSwitch/index.js:4]: This switch looks awesome! Great job

File: src/views/properties/properties.js
Comment (Chris-Boe) [inline src/views/properties/properties.js]: The selectRow functionality seems to be overriding this Link functionality as I can't click into the property details page using the property name

**Review by Chris-Boe — CHANGES_REQUESTED** [review]
Just one issue when trying to click the property name to see the property details page.
After a Slack convo it sounds like we will add the actual archive API calling to this PR after the backend PR is merged

File: src/views/properties/properties.js
Comment (DaveTrost) [reply src/views/properties/properties.js]: This part is going to be weird until we have a view/edit properties page

**Comment by DaveTrost** [comment]
Still waiting for codeforpdx/dwellinglybackend/pull/166 to be merged so this PR can go live. Ping me on Slack if you notice anything (or if this branch needs a rebase)

**Comment by DaveTrost** [comment]
> Still waiting for [codeforpdx/dwellinglybackend/pull/166](https://github.com/codeforpdx/dwellinglybackend/pull/166) to be merged so this PR can go live. Ping me on Slack if you notice anything (or if this branch needs a rebase)

The backend PR is merged. This PR is ready to go! 
@Chris-Boe @htharker42

**Review by Chris-Boe — CHANGES_REQUESTED** [review]
Nice work Dave! I'm just noticing one issue with the search/archive filtering. If I have the Show Archived toggle set to false, type in a search that would match an archived property, then toggle Show Archived to *on*, I still don't see the property that would match my search criteria:
search

File: src/views/properties/properties.js
Comment (DaveTrost) [reply src/views/properties/properties.js]: We have a view/edit properties page - hooray! Thank you @Protopaco!
Clicking on the property name now works as expected. Please take a look @Chris-Boe

**Comment by DaveTrost** [comment]
> Nice work Dave! I'm just noticing one issue with the search/archive filtering. If I have the Show Archived toggle set to false, type in a search that would match an archived property, then toggle Show Archived to _on_, I still don't see the property that would match my search criteria:
> search

Thank you @Chris-Boe for the helpful screen capture! It made it super simple to see what you were highlighting. 
This issue is resolved. Please let me know if you discover anything else ... which you won't because it is now BULLET-PROOF! Muahahaha

**Review by htharker42 — APPROVED** [review]
is working now, good job Dave

**Pull request merged**
